jobs/update-legal-filings/update_legal_filings.py

```diff
@@ -121,24 +120,29 @@ def check_for_manual_filings(application: Flask = None, token: dict = None):
             #       ]
             # }
 
-            # for each event_id: if not in legal db table then add event_id to list
-            for info in colin_events['events']:
-                # check that event is associated with one of the coops loaded into legal db
-                response = requests.get(
-                    f'{legal_url}/{info["corp_num"]}',
-                    headers={'Content-Type': 'application/json', 'Authorization': f'Bearer {token}'}
-                )
-                if response.status_code == 200:
-                    # check legal table
-                    response = requests.get(f'{legal_url}/internal/filings/colin_id/{info["event_id"]}')
-                    if response.status_code == 404:
-                        id_list.append(info)
-                    elif response.status_code != 200:
-                        application.logger.error(f'Error checking for colin id {info["event_id"]} in legal')
-                else:
-                    application.logger.error('No ids returned from colin_last_update table in legal db.')
+    return colin_events['events'] if colin_events else []
 
-    return id_list
+
+def check_for_manual_filings(colin_events, token):
+    """Check for filings."""
+    legal_url = application.config['LEGAL_URL']
+
+    # for each event_id: if not in legal db table then add event_id to list
+    for info in colin_events:
+        # check that event is associated with one of the coops loaded into legal db
+        response = requests.get(
+            f'{legal_url}/{info["corp_num"]}',
+            headers={'Content-Type': 'application/json', 'Authorization': f'Bearer {token}'}
+        )
+        if response.status_code == 200:
+            # check legal table
+            response = requests.get(f'{legal_url}/internal/filings/colin_id/{info["event_id"]}')
+            if response.status_code == 404:
+                yield info
```

Comment:
changed return type to `yield` to process immediately instead of adding it into list and process all together